### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by dron0gus**

### CODE REVIEW

**Comment by dron0gus** [comment]
Execution of `handleCrc32Check` now takes 3mS vs 12mS before.

**Comment by dron0gus** [comment]
No idea why CI fails...

File: firmware/hw_layer/ports/stm32/flash_int.c [hunk 1]
```diff
@@ -82,7 +82,10 @@ static bool intFlashUnlock(void) {
 /**
  * @brief Lock the flash memory for write access.
  */
-#define intFlashLock() { FLASH_CR |= FLASH_CR_LOCK; }
+static void intFlashLock(void)
+{
```
Comment (rusefillc) [inline firmware/hw_layer/ports/stm32/flash_int.c:86]: why is this important?

File: firmware/controllers/can/can_dash.cpp [hunk 2]
```diff
@@ -155,7 +155,8 @@ void canMazdaRX8(CanCycle cycle) {
 
 			float kph = getVehicleSpeed();
 
-			msg.setShortValue(SWAP_UINT16(GET_RPM() * 4), 0);
+			int rpm = GET_RPM();
```
Comment (rusefillc) [inline firmware/controllers/can/can_dash.cpp:158]: i do not understand if this changes everything?
we need code comment or maybe even automated testing coverage if that change affects the outcome somehow

**Comment by rusefillc** [comment]
HW CI is failing because it's failing in master

I've just changed master please rebase

File: firmware/controllers/can/can_dash.cpp (hunk 2 above)
Comment (rusefillc) [reply firmware/controllers/can/can_dash.cpp:158]: I've cherry-picked with a comment please rebase

**Pull request closed without merging**